balancer-js/src/lib/utils/poolHelper.ts

```diff
@@ -143,3 +145,57 @@ export const parsePoolInfo = (
     upScaledBalancesWithoutBpt,
   };
 };
+
+export const parsePoolInfoForProtocolFee = (
```

Comment:
Why don't you simply add the protocolFee logic into the already existing `parsePoolInfo` method?
My feeling is that we should reuse the existing methods as much as possible.